### PR TITLE
docs: Update docs for DELETE support

### PR DESCRIPTION
**Pull request by akshatnair**
DELETE command has some restrictions that needed
documenting.

### CODE REVIEW

File: presto-docs/src/main/sphinx/sql/delete.rst [hunk 1]
```diff
@@ -30,3 +30,9 @@ Delete all line items for low priority orders::
 Delete all orders::
 
     DELETE FROM orders;
+
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/sql/delete.rst:33]: removed this extra line

File: presto-docs/src/main/sphinx/sql/delete.rst [hunk 2]
```diff
@@ -30,3 +30,9 @@ Delete all line items for low priority orders::
...
+
+Limitations
+-------------
+Only :doc:`../connector/hive` currently supports DELETE and it works for the queries where one or more partitions are
+deleted entirely.
```
Comment (kbajda) [inline presto-docs/src/main/sphinx/sql/delete.rst:37]: Is this actually true? Say I have monthly partitions but perform a delete that removes entire year (... WHERE date_year = 2014). That satisfied your description but I think the actual implementation looks at the actual partitioning column and requires that the predicate is just on that column, nothing else. Please double check.

Comment (akshatnair) [inline presto-docs/src/main/sphinx/sql/delete.rst:37]: yes I should clarify that this has to be on partioning column

**Comment by kbajda** [comment]
LGTM

**Pull request closed without merging**